axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceTermServiceImpl.java

```diff
@@ -1046,11 +1046,12 @@ public void setParentFields(
       if (moveLine.getPartner() != null) {
         invoiceTerm.setPartner(moveLine.getPartner());
       } else {
-        invoiceTerm.setPartner(move.getPartner());
+        if (move != null)
+          invoiceTerm.setPartner(move.getPartner());
       }
     }
 
-    if (moveLine != null && invoiceTerm.getOriginDate() == null) {
+    if (move != null && moveLine != null && invoiceTerm != null && invoiceTerm.getOriginDate() == null) {
```

Comment:
I don't think it makes sense to check if invoiceTerm is null here, you can just check it at the start of the method and add a `return;` in that case